### PR TITLE
Refs [TASK][49655] Apply mvvm structure

### PR DESCRIPTION
**Pull request by hungnd-3027**
## Related Tickets
- [#Task 49655: Apply mvvm structure](https://edu-redmine.sun-asterisk.vn/issues/49655)
## WHAT
- apply mvvm structure
## Review Checklist

Category | View Point | Description | Self review | Reviewer2 (name)
--- | --- | --- | --- | ---
Conventions | Sun* coding conventions | https://sal.vn/WtYPAy |<li>- [ ] yes</li>|<li>- [ ] yes</li>
Conventions | Kotlin coding conventions | https://kotlinlang.org/docs/coding-conventions.html |<li>- [ ] yes</li>|<li>- [ ] yes</li>
Redmine | Sun* Redmine working process  | https://sal.vn/piPoQn |<li>- [ ] yes</li>|<li>- [ ] yes</li>

## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

File: app/src/main/java/com/hungngo/themusicapp/base/BaseActivity.kt [hunk 1]
```diff
@@ -0,0 +1,19 @@
+package com.hungngo.themusicapp.base
+
+import androidx.appcompat.app.AppCompatActivity
+import androidx.databinding.DataBindingUtil
+import androidx.databinding.ViewDataBinding
+
+abstract class BaseActivity<B : ViewDataBinding> : AppCompatActivity() {
+    private var _binding: B? = null
+    val binding get() = _binding
```
Comment (vietnh-1109) [inline app/src/main/java/com/hungngo/themusicapp/base/BaseActivity.kt]: cái này để protected hợp lí hơn nhỉ ?

File: app/src/main/java/com/hungngo/themusicapp/data/model/Album.kt [hunk 2]
```diff
@@ -0,0 +1,30 @@
+package com.hungngo.themusicapp.data.model
+
+import com.google.gson.annotations.SerializedName
+
+data class Album (
+    @SerializedName("album_type")
+    val albumType: String,
+    @SerializedName("artists")
+    val artists: List<Artist>,
```
Comment (vietnh-1109) [inline app/src/main/java/com/hungngo/themusicapp/data/model/Album.kt]: mấy biến này để null safety với init cho nó luôn nhé ông ,tránh trg hợp json nó parse sang null

File: app/src/main/java/com/hungngo/themusicapp/utils/extension/AppcompatActivityExtension.kt [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+import androidx.appcompat.app.AppCompatActivity
+import androidx.fragment.app.Fragment
+
+fun AppCompatActivity.addFragment(fragment: Fragment, frameId: Int, backStackTag: String? = null) {
+    supportFragmentManager.beginTransaction().apply {
+        add(frameId, fragment)
+        addToBackStack(backStackTag)
+        commit()
+    }
+}
```
Comment (vietnh-1109) [inline app/src/main/java/com/hungngo/themusicapp/utils/extension/AppcompatActivityExtension.kt]: Tưởng dungf navigation ông :)) dùng cả cái này làm gì

Comment (hungnd-3027) [reply app/src/main/java/com/hungngo/themusicapp/utils/extension/AppcompatActivityExtension.kt]: mấy cái dependency là tôi cop từ project cũ mà chưa bỏ nó đi @@

**Review by vietnh-1109 — APPROVED** [review]

**Pull request merged**